iac-extensions/arm/src/main/java/org/sonar/iac/arm/tree/impl/bicep/SyntaxTokenImpl.java

```diff
@@ -57,4 +57,8 @@ public List<Tree> children() {
   public Kind getKind() {
     throw new UnsupportedOperationException("No kind for SyntaxToken");
   }
+
+  public void setTextRange(TextRange textRange) {
```

Comment:
In the current implementation, it is used to shift ranges of nodes of Bicep expression according to the location, where it is present in JSON.

Reply:
I am not a fan of changing TextRanges afterward. Typically we create a new SyntaxToken to be able to process the original one.

Reply:
Yes, I agree. However, we don't have a straightforward way to modify children of an existing node as well. The idea here is that we build a Bicep tree from a string, and the nodes will start at `1:0`. Then we need to shift their start offset to point to the beginning of the expression string. This would mean traversing the Bicep tree and replacing all `SyntaxToken`s with shifted copies , but for that we don't have a method in `Tree`.

Reply:
And actually a similar problem arises in the next ticket when I need to substitute `FunctionCall` nodes with `Identifier`s for parameters and variables. Both this and shifting of SyntaxTokens can be solved by e.g. introducing a `replace(Tree old, Tree new)` in the `ArmTree`. However, I'm not sure if it's the best approach.

Reply:
I think we can now get rid of this function.